gensim/models/phrases.py

```diff
@@ -169,7 +169,9 @@ def learn_vocab(sentences, max_vocab_size, delimiter=b'_', progress_per=10000):
             if sentence_no % progress_per == 0:
                 logger.info("PROGRESS: at sentence #%i, processed %i words and %i word types" %
                             (sentence_no, total_words, len(vocab)))
-            sentence = [utils.any2utf8(w) for w in sentence]
+
+            sentence = [w for w in (utils.any2utf8(u'_'.join(sentence)).split('_'))]
```

Comment:
A few issues here - 
1. You are trying to split a bytestring (the result of the `any2utf8` call) by `'_'` - this will not work in python3+ because literal strings are unicode by default. You've faced similar problems previously, so I think it would be helpful to understand character encodings at a conceptual level, and the differences between string handling in python2 and 3.

2. Simply `sentences = utils.any2utf8(u'_'.join(sentence)).split('_'))` would be enough - no need for the extra `[w for w in ...]`

3. We're not accounting for the possibility that a word in the sentence contains `'_'` here - it would be wrong to make implicit assumptions like these about user input, unless there was an explicit constraint in the API. Escaping could be an option - although I'm not sure it is feasible, performance-wise.